src/app/ReadHandler.cpp

```diff
@@ -773,7 +773,7 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
 
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
 
-    // Default behavior for ICDs where the wanted MaxInterval for a subscription is the IdleModeDuration
+    // Default behavior for LIT ICDs where the wanted MaxInterval for a subscription is the IdleModeDuration
```

Comment:
This probably needs more reasoning based on the discussion below. It seems what we want to consider is:
 - firmware updates to LIT support (i.e. they start including the ICD Management cluster) but _before_ the device is configured as a LIT
 
It seems we want to describe that:
  - we want ICD Management cluster to take effect if device is in LIT mode
  - if in SIT, we skip any ICD management on purpose because the ecosystem may not yet be capable to handle
a long sleep device

Did I understand the issue correctly @yunhanw-google ?

@lpbeliveau-silabs what is the behaviour you would expect? I believe the concern is that "new OTA brings in LIT ICD support, but we don't want to break ecosystems that are not yet supporting LIT-ICD".

Reply:
yes, more specifically, code in readHandler covered by CHIP_CONFIG_ENABLE_ICD_SERVER should not impact the existing SIT behavior.

Reply:
There seems to be some misunderstanding here, I'd like to discuss what is the issue we are trying to fix and because I don't think I understand correctly from the description.
Can we sync on Monday?